doc/CMakeLists.txt

```diff
@@ -148,10 +149,10 @@ add_custom_target(
                 "External Module,modules,${ZEPHYR_BASE}/modules"
 
   VERBATIM
-  WORKING_DIRECTORY ${CMAKE_CURRENT_LIST_DIR}
-  COMMENT "Running gen_kconfig_rest.py ${DOCS_SRC_DIR}/reference/kconfig/"
 )
 
+set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${GEN_KCONFIG_REST_SCRIPT})
```

Comment:
Maybe add info in the commit message about this.

Appending `_scripts/gen_kconfig_rest.py` is quite an important piece of information, as that ensures that CMake will re-run when the python script is updated.